packages/mongodb-constants/src/bson-types.ts

```diff
@@ -122,6 +122,46 @@ const BSON_TYPES = [
     description: 'BSON Regex type',
     snippet: "RegExp('${1:source}', '${2:opts}')",
   },
+  {
+    name: 'LegacyJavaUUID',
+    value: 'LegacyJavaUUID',
+    label: 'LegacyJavaUUID',
+    score: 1,
+    meta: 'bson-legacy-uuid',
```

Comment:
The `meta` here will make it into Compass' autocompleter:

<img width="756" height="242" alt="Image" src="https://github.com/user-attachments/assets/d7e34dab-6860-4d96-8ec9-b8f0c5386d97" />